spacy/pipeline/entity_linker.py

```diff
@@ -74,11 +85,18 @@ def make_entity_linker(
     incl_prior: bool,
     incl_context: bool,
     entity_vector_length: int,
-    get_candidates: Callable[[KnowledgeBase, Span], Iterable[Candidate]],
+    get_candidates: Callable[
+        [BaseKnowledgeBase, Union[Span, str]], Iterator[Candidate]
```

Comment:
Strictly speaking we expect `get_candidates()` to return `Union[Iterator[Candidate], Iterable[Candidate]]`, right? As long as it's convertable to a list, it should be fine.

Reply:
I think the typing was wrong before? It should have been `Iterable`, no?

Reply:
The docs actually have it correct, `Iterable`! So we can change this as a "bug fix".